src/Core/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -63,27 +65,144 @@ public string Build(SqlQueryStructure structure)
+                // actual table and the temporary table based on the values of the non-autogenerated PK columns.
+                // If there is a column in the PK which is autogenerated, we cannot and will not insert it into the temporary table.
+                // Hence in the select query, we will add an additional WHERE predicate (alongwith the join).
+                // for the autogenerated column in the PK.
+                // It is to be noted that MsSql supports only one IDENTITY/autogenerated column per table.
+
+                string tempTableName = $"{QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier($"#{structure.DatabaseObject.Name}_T")}";
+                (string autoGenPKColumn, List<string> nonAutoGenPKColumns) = GetSegregatedPKColumns(sourceDefinition);
+                if (nonAutoGenPKColumns.Count > 0)
+                {
+                    // Create temporary table containing zero rows and all the non-autogenerated columns present in the PK.
+                    // We need to create it only when there are non-autogenerated columns present in the PK.
+                    string queryToCreateTempTable = $"SELECT {string.Join(", ", nonAutoGenPKColumns.Select(pk => $"{QuoteIdentifier(pk)}"))}" +
+                        $" INTO {tempTableName} FROM {tableName} WHERE 0 = 1;";
+
+                    // We need to output values of all the non-autogenerated columns in the PK into the temporary table.
+                    string nonAutoGenPKsOutput = string.Join(", ", nonAutoGenPKColumns.Select(pk => $"{OutputQualifier.Inserted}.{QuoteIdentifier(pk)}"));
+
+                    // Creation of temporary table followed by inserting data into actual table.
+                    insertQuery.Append(queryToCreateTempTable);
+                    insertQuery.Append($"INSERT INTO {tableName} ({insertColumns}) ");
+                    insertQuery.Append($"OUTPUT {nonAutoGenPKsOutput} INTO {tempTableName} ");
+                }
+                else
+                {
+                    insertQuery.Append($"INSERT INTO {tableName} ({insertColumns}) ");
+                }
+
+                insertQuery.Append(values);
+
+                // Build the subsequent select query to return the inserted data.
+                StringBuilder subsequentSelect = new($"SELECT {MakeOutputColumns(structure.OutputColumns, tableName)} FROM {tableName} ");
+
+                if (nonAutoGenPKColumns.Count > 0)
+                {
+                    // We will perform inner join on the basis of all the non-autogenerated columns in the PK.
+                    string joinPredicates = string.Join(
+                        "AND ",
+                        nonAutoGenPKColumns.Select(pk => $"{tableName}.{QuoteIdentifier(pk)} = {tempTableName}.{QuoteIdentifier(pk)}"));
+                    subsequentSelect.Append($"INNER JOIN {tempTableName} ON {joinPredicates} ");
```

Comment:
> (i see the next block of code appends an additional WHERE clause using scope_identity() so maybe that answer my question.

yes, this will help uniquely get a record.